Add Documentation (#12)
trento-project/telemetry: docs/telemetry.md

# **Telemetry**

>Telemetry is the in situ collection of measurements or other data at remote points and their automatic transmission to receiving equipment (telecommunication) for monitoring.
>
[Telemetry, Wikipedia](https://en.wikipedia.org/wiki/Telemetry)

---
## **Pitch**

At SUSE we believe in [continuous improvement](https://en.wikipedia.org/wiki/Kaizen), and we want our products get better over time and as our customers' needs change.

In this process we will have to make decisions, and we believe that a wide range of *inputs*, *perspectives* and *information* will help us in making better decisions.

This is why, next to a close feedback-based collaboration with our customers, we are coming with this **Telemetry Service**, which is responsible of collecting meaningful inputs for our improvement.

Telemetry is used by SUSE to make the products better and is not shared with third-parties.

---
## **Concepts**

> ### **Telemetry**

Reviewer: ```suggestion
### **Telemetry Data**
```

# **Telemetry**

>Telemetry is the in situ collection of measurements or other data at remote points and their automatic transmission to receiving equipment (telecommunication) for monitoring.
>
[Telemetry, Wikipedia](https://en.wikipedia.org/wiki/Telemetry)

---
## **Pitch**

At SUSE we believe in [continuous improvement](https://en.wikipedia.org/wiki/Kaizen), and we want our products get better over time and as our customers' needs change.

In this process we will have to make decisions, and we believe that a wide range of *inputs*, *perspectives* and *information* will help us in making better decisions.

This is why, next to a close feedback-based collaboration with our customers, we are coming with this **Telemetry Service**, which is responsible of collecting meaningful inputs for our improvement.

Telemetry is used by SUSE to make the products better and is not shared with third-parties.

---
## **Concepts**

### **Telemetry**
Anonymous information collected to provide extra feedbacks to the organization for strategic decisions towards improvement.

> eg. *number of CPUs*, *SLES version*, *Cloud Provider*

Reviewer: As defined above, Telemetry is actually the _action_ of collecting such information, not the information itself.

Maybe it's worth removing this sub-paragraph altogether; it's kinda duplicated with the opening section. Unless here you wanted to discuss the single pieces of data specifically, then we need to completely reword this.

Author: > As defined above, Telemetry is actually the action of collecting such information, not the information itself.
>
Correct. Here I wanted to give a short definition of the pieces of data being collected.
**Telemetry** is indeed not really appropriate, how can we rephrase it?
- **Telemetry Data**
- **Collected Data**
- **Data**
- others?

Reviewer: `Telemetry Data` should be ok

### **Source**
A Source is possibly any of the products/services that [SUSE offers](https://www.suse.com/products/) which can be enabled to publish telemetry.

For instance 
- Trento
- SUSE Linux Enterprise Server
- SUSE Linux Enterprise Server for SAP Applications
- others...

Every *Source* is free to define its own telemetry and responsible to implement its *Extraction* and subsequent *Publishing* to the Telemetry Service.

### **Telemetry Service**

> Telemetry Service (aka *Collector Service*, *Telemetry Collector Service*) is an application whose main responsibility is to collect meaninguful **Metrics** being published by different **Sources**.

Reviewer: I'd say we don't collect just metrics, we collect data in general. A "metric" is by definition a measurement expressed by  an numerical value.
```suggestion
Telemetry Service (aka *Collector Service*, *Telemetry Collector Service*) is an application whose main responsibility is to collect meaningful **Data** being published by different **Sources**.
```

and *process* (?) that information in order to make it Consumption-ready.

### **Consumption**
T.B.D.

This is a *deliberatedly-very-generic* placeholder for what comes after collection.

That is the organization consuming collected information as feedbacks for product improvement.

> Out of scope atm.

Reviewer: ```suggestion
### **Consumption**

The consumption of the collected data is out of the scope of this service at the moment, but the core tenet stands that said consumption is only done with the purpose of establishing a closed feedback loop to improve our products.
```
